governator-core/src/main/java/com/netflix/governator/LifecycleInjectorCreator.java

```diff
@@ -0,0 +1,152 @@
+        this.profiles = new LinkedHashSet<>(profiles);
+        return this;
+    }
+    
+    public LifecycleInjectorCreator withFeatures(IdentityHashMap<GovernatorFeature<?>, Object> features) {
+        this.features = features;
+        return this;
+    }
+
+    @Singleton
+    @SuppressLifecycleUninitialized
+    class GovernatorFeatureSetImpl implements GovernatorFeatureSet {
+        private final IdentityHashMap<GovernatorFeature<?>, Object> featureOverrides;
+        
+        @Inject
+        private PropertySource properties = new DefaultPropertySource();
+        
+        @Inject
+        public GovernatorFeatureSetImpl(IdentityHashMap<GovernatorFeature<?>, Object> featureOverrides) {
+            this.featureOverrides = featureOverrides;
+        }
+        
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> T get(GovernatorFeature<T> feature) {
+            return featureOverrides.containsKey(feature)
+                ? (T) featureOverrides.get(feature)
+                : (T) properties.get(feature.getKey(), feature.getType(), feature.getDefaultValue());
+        }
+    }
+
+    @Override
+    public LifecycleInjector createInjector(Stage stage, Module module) {
+        final GovernatorFeatureSetImpl featureSet = new GovernatorFeatureSetImpl(features);
+        
+        final LifecycleManager manager = new LifecycleManager();
+        
+        // Construct the injector using our override structure
+        try {
+            onBeforeInjectorCreated();
+            Injector injector = Guice.createInjector(
+                stage, 
+                module,
+                new LifecycleModule(),
+                new LegacyScopesModule(),
+                new AbstractModule() {
+                    @Override
+                    protected void configure() {
+                        bind(GovernatorFeatureSet.class).toInstance(featureSet);
+                        bind(LifecycleManager.class).toInstance(manager);
+                        bind(new TypeLiteral<Set<String>>() {}).annotatedWith(Profiles.class).toInstance(profiles);
+                        bind(String[].class).annotatedWith(Arguments.class).toInstance(args);
+                        requestInjection(LifecycleInjectorCreator.this);
+                    }
+                });
+            manager.notifyStarted();
+            return LifecycleInjector.wrapInjector(injector, manager);
+        }
+        catch (ProvisionException|CreationException|ConfigurationException e) {
+            LOG.error("Failed to create injector", e);
+            try {
+                manager.notifyStartFailed(e);
+            }
+            catch (Exception e2) {
+                LOG.error("Failed to notify injector creation failure", e2 );
+            }
+            if (!featureSet.get(GovernatorFeatures.SHUTDOWN_ON_ERROR)) {
+                return LifecycleInjector.createFailedInjector(manager);
+            }
+            else {
+                throw e;
```

Comment:
The finally's onAfterInjectorCreated() will still apply for this exception. Since this is the fatal case for create injector, we should not execute onAfterInjectorCreated().